code/week6/main.cpp

```diff
@@ -0,0 +1,58 @@
+#include <opencv2/opencv.hpp>
+#include <string.h>
+#include <stdlib.h>
+
+int main() {
+    using namespace std;
+
+    // Mat http://docs.opencv.org/3.3.0/d3/d63/classcv_1_1Mat.html
+    // this creates an image header. The array is not allocated yet
+    cv::Mat my_image;
+
+    /* This creates a matrix
+     * cv::Mat(rows, cols, type, default color)
+     * type: CV_[The number of bits per item][Signed or Unsigned][Type Prefix]C[The channel number]
+     * this type CV_8UC3
+     * 8 is the number of bits per item
+     * U means that it is unsigned
+     * C3 means that the color channel has three entires (BGR)
+    */
+    my_image = cv::Mat(2, 2, CV_8UC3, cv::Scalar(0,0,255));
+
+    // a sinlge pixel is made up of 3 8 bit values
+    // Blue, Green, Red
+    cout << "my_image = \n" << my_image << endl;
+
+    // iterate over an image
+    for(auto it = my_image.begin<cv::Vec3b>(); it != my_image.end<cv::Vec3b>(); it++) {
```

Comment:
This is not the fastest way to iterate over an image in OpenCV. Do we care?

Reply:
do we want an example of iterating over a Mat or should this just be taken out?

Reply:
That's up to someone else. It's a useful thing to know, but you can also do a lot without using it.

Reply:
I added something along the lines of this http://docs.opencv.org/2.4/doc/tutorials/core/how_to_scan_images/how_to_scan_images.html#howtoscanimagesopencv